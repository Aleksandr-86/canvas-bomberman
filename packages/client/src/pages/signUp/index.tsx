const SignUp = () => {
  return <h2>SignUp page</h2>
}

export { SignUp }
