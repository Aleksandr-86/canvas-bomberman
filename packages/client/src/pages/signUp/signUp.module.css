.signUp {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 900px;
  padding: 10px 0;
  margin: 30px 0;
}

.signUpHeader {
  font-size: 40px;
}

.signUpContent {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  width: 100%;
  margin-top: 100px;
}

.signUpAction {
  display: flex;
  flex-direction: inherit;
  align-items: center;
  width: 100%;
  margin-top: 100px;
}

.actionButton {
  margin-bottom: 30px;
}
