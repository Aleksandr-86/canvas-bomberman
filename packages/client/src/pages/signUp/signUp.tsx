import React, { useEffect, useState } from 'react'

import { Link, useNavigate } from 'react-router-dom'
import { Button } from '../../components/button/button'
import { FormField } from '../../components/formField/formField'

import { useForm, Validations } from '../../hooks/useForm'
import { getValidations } from '../../features/validation'

import { useAppDispatch } from '../../store/hooks'
import { registerUser } from '../../store/userActions'
import { transformUserDTO } from '../../features/api/apiTransformers'

import styles from './signUp.module.css'
import baseStyles from '../../app/app.module.css'

type FormType = {
  login: string
  firstName: string
  secondName: string
  password: string
  repeatPassword: string
  phone: string
  email: string
}

export const SignUp: React.FC = () => {
  const navigate = useNavigate()
  const dispatch = useAppDispatch()
  const [validations, setValidations] = useState<Validations<FormType>>()

  useEffect(() => {
    setValidations(
      getValidations([
        'login',
        'firstName',
        'secondName',
        'phone',
        'email',
        'password',
      ])
    )
  }, [])

  const { values, errors, register, watch, isValid } = useForm<FormType>({
    validations: {
      ...validations,
      repeatPassword: {
        custom: {
          validation: (value: string) => value === watch('password'),
          message: `Пароли не совпадают`,
        },
      },
    },
  })

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault()

    if (isValid()) {
      const { repeatPassword, ...data } = values

      dispatch(registerUser(transformUserDTO(data)))
      navigate('/')
    }
  }

  return (
    <>
      <form className={styles.signUp} onSubmit={handleSubmit}>
        <div className={styles.signUpHeader}>Регистрация</div>

        <div className={styles.signUpContent}>
          <FormField
            label="Логин"
            name="name"
            type="text"
            {...register('login')}></FormField>
          <FormField
            label="Имя"
            name="firstName"
            type="text"
            {...register('firstName')}></FormField>
          <FormField
            label="Фамилия"
            name="secondName"
            type="text"
            {...register('secondName')}></FormField>
          <FormField
            label="Почта"
            name="email"
            type="email"
            {...register('email')}></FormField>
          <FormField
            label="Телефон"
            name="phone"
            type="tel"
            {...register('phone')}></FormField>
          <FormField
            label="Пароль"
            name="password"
            type="password"
            {...register('password')}></FormField>
          <FormField
            label="Пароль еще раз"
            name="repeatPassword"
            type="password"
            {...register('repeatPassword')}></FormField>
        </div>

        <div className={styles.signUpAction}>
          <div className={styles.actionButton}>
            <Button type="submit">Все верно!</Button>
          </div>

          <div className={styles.actionButton}>
            <Link className={baseStyles.linkButton} to="/">
              На Главную
            </Link>
          </div>
        </div>
      </form>
    </>
  )
}
