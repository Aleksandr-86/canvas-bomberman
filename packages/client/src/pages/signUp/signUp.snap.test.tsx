import * as renderer from 'react-test-renderer'
import { SignUp } from './signUp'

describe(`<SignUp/> snapshot test cases`, () => {
  test(`<SignUp/> should render <h2>SignUp page</h2>`, () => {
    const tree = renderer.create(<SignUp />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
