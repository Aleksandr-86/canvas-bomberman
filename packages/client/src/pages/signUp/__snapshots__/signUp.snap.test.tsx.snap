// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp/> snapshot test cases <SignUp/> should render <h2>SignUp page</h2> 1`] = `
<h2>
  SignUp page
</h2>
`;
